pandas/tests/groupby/test_groupby.py

```diff
@@ -1822,11 +1830,36 @@ def test_empty_groupby(columns, keys, values, method, op, request):
     df = df.iloc[:0]
 
     gb = df.groupby(keys)[columns]
-    if method == "attr":
-        result = getattr(gb, op)()
-    else:
-        result = getattr(gb, method)(op)
 
+    def get_result():
+        if method == "attr":
```

Comment:
this is a super complicated test. can you try to split in followons.

Reply:
agreed